packages/flutter_tools/lib/src/cache.dart

```diff
@@ -1058,6 +1059,58 @@ class FlutterRunnerSDKArtifacts extends CachedArtifact {
   }
 }
 
+/// Implementations of this class can resolve URLs for packages that are versioned.
+///
+/// See also [CipdArchiveResolver].
+abstract class VersionedPackageResolver {
+  /// Returns the URL for the artifact.
+  String resolveUrl(String packageName, String version);
+}
+
+/// Resolves the CIPD archive URL for a given package and version.
+class CipdArchiveResolver extends VersionedPackageResolver {
+  @override
+  String resolveUrl(String packageName, String version) {
+    return '$_cipdBaseUrl/flutter/$packageName/+/git_revision:$version';
+  }
+}
+
+/// The debug symbols for flutter runner for Fuchsia development.
+class FlutterRunnerDebugSymbols extends CachedArtifact {
+  FlutterRunnerDebugSymbols(Cache cache, this.packageResolver, {this.dryRun = false})
```

Comment:
With the const constructor above you can make packageResolver a named argument that defaults to `const FlutterRunnerDebugSymbols`

Reply:
Good idea. Done.